fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -456,6 +458,34 @@ private FHIRRestOperationResponse doPatchOrUpdate(String type, String id, FHIRPa
                 } catch (FHIRPersistenceResourceDeletedException e) {
                     isDeleted = true;
                 }
+
+                if (skippableUpdate && !isDeleted) {
+                    ResourceFingerprintVisitor fingerprinter = new ResourceFingerprintVisitor();
+                    ior.getPrevResource().accept(fingerprinter);
+                    SaltHash baseline = fingerprinter.getSaltAndHash();
+
+                    fingerprinter = new ResourceFingerprintVisitor(baseline);
+                    newResource.accept(fingerprinter);
+                    if (fingerprinter.getSaltAndHash().equals(baseline)) {
+                        txn.commit();
+                        txn = null;
+
+                        ior.setResource(ior.getPrevResource());
+                        ior.setStatus(Status.OK);
+                        ior.setLocationURI(FHIRUtil.buildLocationURI(type, ior.getPrevResource()));
+                        ior.setOperationOutcome(OperationOutcome.builder()
+                                .issue(Issue.builder()
+                                    .severity(IssueSeverity.INFORMATION)
+                                    .code(IssueType.INFORMATIONAL)
+                                    .details(CodeableConcept.builder()
+                                        .text(string("Update resource matches the existing resource; skipping the update"))
+                                        .build())
+                                    .build())
+                                .build());
+
+                        return ior; // early exit
+                    }
+                }
```

Comment:
this is the meat of the change

Reply:
sorry, I know I'm not a reviewer, what happens in the Audit Path?  what data is sent on the Audit? is it consistent?

Reply:
great question.  presently it will emit an audit log just like any other update, except that the prevResource will match the newResource.
```
                RestAuditLogger.logUpdate(httpServletRequest,
                        ior != null ? ior.getPrevResource() : null,
                        ior != null ? ior.getResource() : null,
                        startTime, new Date(), status);
```

Reply:
let me know if you think we need to do something different

Reply:
then I think we're good... it was good to confirm.  thanks!